src/recommendation/src/main/python/RankingTrainValidationSplit.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (C) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in the project root for information.
+
+import sys
+from mmlspark.RankingTrainValidationSplitModel import RankingTrainValidationSplitModel as tvmodel
+from mmlspark._RankingTrainValidationSplit import _RankingTrainValidationSplit
+from pyspark import keyword_only
+from pyspark.ml.param import Params
+from pyspark.ml.tuning import ValidatorParams
+from pyspark.ml.util import *
+
+if sys.version >= '3':
+    basestring = str
+
+
+@inherit_doc
+class RankingTrainValidationSplit(_RankingTrainValidationSplit, ValidatorParams):
+    @keyword_only
+    def __init__(self, estimator=None, estimatorParamMaps=None, evaluator=None, seed=None):
+        """
+        __init__(self, estimator=None, estimatorParamMaps=None, evaluator=None, numFolds=3,\
+                 seed=None)
+        """
+        super(RankingTrainValidationSplit, self).__init__()
+        kwargs = self._input_kwargs
+        self._set(**kwargs)
+
+    @keyword_only
+    def setParams(self, estimator=None, estimatorParamMaps=None, evaluator=None, seed=None):
+        """
+        setParams(self, estimator=None, estimatorParamMaps=None, evaluator=None, numFolds=3,\
+                  seed=None):
+        Sets params for cross validator.
+        """
+        kwargs = self._input_kwargs
+        return self._set(**kwargs)
+
+    def copy(self, extra=None):
+        """
+        Creates a copy of this instance with a randomly generated uid
+        and some extra params. This copies creates a deep copy of
+        the embedded paramMap, and copies the embedded and extra parameters over.
+
+        :param extra: Extra parameters to copy to the new instance
+        :return: Copy of this instance
+        """
+        if extra is None:
+            extra = dict()
+        newCV = Params.copy(self, extra)
+        if self.isSet(self.estimator):
+            newCV.setEstimator(self.getEstimator().copy(extra))
+        # estimatorParamMaps remain the same
+        if self.isSet(self.evaluator):
+            newCV.setEvaluator(self.getEvaluator().copy(extra))
+        return newCV
+
+    def _create_model(self, java_model):
```

Comment:
if you are not overriding the model, which it looks like you are not, then this is already handled in the base class right?

Reply:
oh, it looks like you put the model in a separate file and you are overriding it.  I think we usually put them in the same file.